tests/foreman/cli/test_leapp.py

```diff
@@ -0,0 +1,210 @@
+    'rhel8_8_bos': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)',
+    'rhel8_8_aps': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)',
+    'rhel9_2_bos': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)',
+    'rhel9_2_aps': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)',
+}
+CUSTOM_REPOS = {
+    'rhel8_8_bos': {
+        'id': 'rhel-8-for-x86_64-baseos-rpms',
+        'name': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS RPMs 8.8',
+        'releasever': '8.8',
+        'version': '8',
+        'reposet': CUSTOM_REPOSET['rhel8_8_bos'],
+        'product': PRDS['rhel8'],
+        'distro': 'rhel8',
+        'key': 'rhel8_bos',
+        'basearch': 'x86_64',
+    },
+    'rhel8_8_aps': {
+        'id': 'rhel-8-for-x86_64-appstream-rpms',
+        'name': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream RPMs 8.8',
+        'releasever': '8.8',
+        'basearch': 'x86_64',
+        'version': '8',
+        'reposet': CUSTOM_REPOSET['rhel8_8_aps'],
+        'product': PRDS['rhel8'],
+        'distro': 'rhel8',
+        'key': 'rhel8_aps',
+    },
+    'rhel9_2_bos': {
+        'id': 'rhel-9-for-x86_64-baseos-rpms',
+        'name': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS RPMs 9.2',
+        'releasever': '9.2',
+        'version': '9',
+        'reposet': CUSTOM_REPOSET['rhel9_2_bos'],
+        'product': PRDS['rhel9'],
+        'distro': 'rhel9',
+        'key': 'rhel9_bos',
+        'basearch': 'x86_64',
+    },
+    'rhel9_2_aps': {
+        'id': 'rhel-9-for-x86_64-appstream-rpms',
+        'name': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream RPMs 9.2',
+        'releasever': '9.2',
+        'basearch': 'x86_64',
+        'version': '9',
+        'reposet': CUSTOM_REPOSET['rhel9_2_aps'],
+        'product': PRDS['rhel9'],
+        'distro': 'rhel9',
+        'key': 'rhel9_aps',
+    },
+}
+
+
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('8')
+def test_upgrade_rhel8_to_rehl9(
+    target_sat,
+    default_org,
+    rhel_contenthost,
+):
+    """
+    Test to upgrade RHEL host to next major RHEL Realse with Leapp Preupgrade and Leapp Upgrade
+    Job templates
+    :Steps:
+        1. Import a subscription manifest and enable, sync source & target repositories
+        2. Create CV, add repositories, create lce, ak, etc.
+        3. Register contenthost using ak
+        4. Varify target rhel repositories are enable on Satellite
+        5. Update all packages, install leapp tool and fix inhibitors
+        6. Run Leapp Preupgrade and Leapp Upgrade job template
+    :ExpectedResult:
+        1. Update RHEL OS major version to another major version
+    """
+    # 1. Import a subscription manifest
+    target_sat.upload_manifest(default_org)
+    # Enable rhel8/rhel9 bos, aps repository and add in content view
+    lc_env = target_sat.api.LifecycleEnvironment(organization=default_org).create()
+    c_view = target_sat.api.ContentView(organization=default_org).create()
```

Comment:
First 3 steps can be part of setup fixture you can create and move it there as it will be common for all the scenarios

Reply:
added new fixture